[AIRFLOW-157][AIRFLOW-175] Add Python3 compatibility and JIRA command to PR tool

jlowin opened the pull request:
The auto-JIRA part of the PR tool was not Python 3 compatible (lots of indexing filter objects that would have been lists in Python 2).

In addition, this creates a separate `close_jira` command that can be used to close JIRA issues without merging a PR.

Also, it fixes an issue where the JIRA issue inference expected a 4-digit number.

https://issues.apache.org/jira/browse/AIRFLOW-157

jlowin committed:
[AIRFLOW-157] Make PR tool Py3-compat; add JIRA command

- Adds Python3 compatibility (filter objects can't be indexed)
- Adds JIRA command to close issues without merging a PR
- Adds general usability fixes and starts cleaning up code

criccomini commented on dev/airflow-pr:
```diff
@@ -53,7 +54,7 @@ except ImportError:
     import click
 except ImportError:
-    print("Could not find the click library. Run 'sudo pip install click' to install.")
+    click.echo("Could not find the click library. Run 'sudo pip install click' to install.")
```
This is recursive isn't it? ;)

jlowin committed:
[AIRFLOW-175] Run git-reset before checkout in PR tool

If the user made any changes, git checkout will fail because the
changes would be overwritten. Running git reset blows the changes away.
